architectureDecisionRecords/0003-monorepo.md

```diff
@@ -0,0 +1,31 @@
+# Title
+
+Monorepo
+
+## Context
+
+In its current state setting up a local development environment for UCP is difficult and confusing with so many repositories. To develop locally we currently have to run multiple different projects, and configure them to work together.
+
+There isn't a standard folder structure between repositories. This repository has a confusing folder structure where the root package.json file relates to the server, and the ui folder has its own package.json.
+
+## Decision
+
+We will convert this repository into a monorepo, and merge all repositories that are necessary to run UCP into this repository.
+
+We will use [Turborepo](https://turbo.build/repo/docs) for our monorepo tooling.
+
+All packages will be installed with a single npm install command from root.
+
+All services will start with a single command from root.
+
+Each service will have its own .env file.
+
+## Consequences
+
+Contributors will need to have a basic working knowledge of turborepo to contribute.
+
+Making changes will be easier, because you won't need to put PRs up for multiple repositories when they need changes to be able to work together.
+
+Shared packages will be easy to create, and they won't require publishing to npm.
```

Comment:
One point that Emily pointed out, is that the widget needs to be retained as a separate package so that companies that want to customize the widget won't need to do so from the monorepo. MX's use case is an example of this. 

Because of this, I'm not sure of the best way to incorporate the NPM widget repo into this scenario.

Reply:
We discussed a bit in standup, but not all shared packages have to live here. I don't think the widget should live here, but there will be other small packages of shared code that should.

Reply:
One thing that Jishun has brought up, is that the way we use the NPM package currently is pretty cumbersome for customization, and even for updates. Maybe we can brainstorm how that can be improved.